### PR TITLE
refactor(backend): switch from warp to hyper

### PR DESCRIPTION
**Pull request by ravenclaw900**
Switches the webserver framework from Warp (used since I rewrote the dashboard in Rust) to Hyper (the lower-level framework behind Warp). Reduces size by ~.4 MB, decreases compile time, and probably increases speed and response time when running the dashboard, though I haven't tested this yet.

### CODE REVIEW

**Comment by MichaIng** [comment]
Nice, I'm all in for switching from high-level to low-level frameworks, especially when like in this case the overall code lines are just slightly increased and source code complexity not really increased.

But probably a benchmark would be interesting to see whether warp does some smart tweaks behind the scenes we'd need to apply explicitly with hyper.

**Comment by ravenclaw900** [comment]
Simple benchmark done, the results are fairly self-explanatory.
Warp:
```
wrk --latency -t4 -c200 -d8s http://localhost:5252
Running 8s test @ http://localhost:5252
  4 threads and 200 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   124.59ms   67.66ms 237.55ms   51.92%
    Req/Sec   367.70    194.60     1.61k    83.39%
  Latency Distribution
     50%  113.41ms
     75%  195.11ms
     90%  209.59ms
     99%  224.10ms
  11706 requests in 8.01s, 8.53MB read
  Socket errors: connect 0, read 973, write 45, timeout 0
Requests/sec:   1460.61
Transfer/sec:      1.06MB
```
Hyper:
```
wrk --latency -t4 -c200 -d8s http://localhost:5252
Running 8s test @ http://localhost:5252
  4 threads and 200 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    38.99ms   17.39ms  90.54ms   79.33%
    Req/Sec     1.19k   373.69     2.39k    61.44%
  Latency Distribution
     50%   34.58ms
     75%   41.51ms
     90%   74.16ms
     99%   83.26ms
  37976 requests in 8.01s, 32.09MB read
  Socket errors: connect 0, read 688, write 55, timeout 0
Requests/sec:   4741.33
Transfer/sec:      4.01MB
```

**Comment by MichaIng** [comment]
Oh great, yes pretty significant, no need for me to spin up some Python HTTP benchmarking script I've still around when benchmarking some changes in [cheroot](https://pypi.org/project/cheroot/) 😄. [wrk](https://packages.debian.org/bullseye/wrk), something to keep in mind 👍.

File: src/main.rs [hunk 1]
```diff
@@ -2,261 +2,60 @@
...
-        warp::serve(routes)
-            .tls()
-            .cert_path(&CONFIG.cert)
-            .key_path(&CONFIG.key)
-            .run((addr, CONFIG.port))
-            .await;
+        simple_hyper_server_tls::hyper_from_pem_files(
+            &CONFIG.cert,
+            &CONFIG.key,
+            simple_hyper_server_tls::Protocols::HTTP1,
```
Comment (MichaIng) [inline src/main.rs:39]: Any reason to not use:
```suggestion
            simple_hyper_server_tls::Protocols::ALL,
```
so clients can choose the HTTP protocol version?

Comment (ravenclaw900) [reply src/main.rs:39]: At the moment, to cut down on size, only HTTP/1.1 support is compiled in the binary (https://github.com/ravenclaw900/DietPi-Dashboard/blob/hyper/Cargo.toml#L27, https://github.com/ravenclaw900/DietPi-Dashboard/blob/hyper/Cargo.toml#L29)

Comment (MichaIng) [reply src/main.rs:39]: Okay, actually makes sense as long as we do not make explicit use of any HTTP/2 features.

**Review by MichaIng — DISMISSED** [review]

**Comment by ravenclaw900** [comment]
Alright, switched to our own TLS setup (still using `rustls`, but encrypting the connection on our own), which lets us return the remote peer address, which lets us show it in tracing spans (which I wanted to do with Warp, but couldn't). Should be ready to merge now.

**Pull request merged**